### PR TITLE
Add regexp for better error handling in addauthor (orcide, inst, email)

### PR DESCRIPTION
**Pull request by jwbos**

### CODE REVIEW

**Comment by jwbos** [comment]
Do we need more checks or tests?

**Review by kmccurley — COMMENTED** [review]
For now I'll leave this as comments.

File: iacrcc/iacrcc.cls [hunk 1]
```diff
@@ -552,6 +552,44 @@
...
+{
+  \cs_generate_variant:Nn \regex_match:nnTF {nV}
+  % An institute index should be a number, multiple institutes are separated by a comma:
+  \regex_match:nnTF {^[,\d]+$} { #1 }
+  { \prg_return_true:  }
+  { \prg_return_false: }
+}
+\cs_new_eq:NN \IfValidInst \iacr_check_inst:nTF
+
+\prg_new_conditional:Npnn \iacr_check_email:n #1 { T, F, TF }
```
[the remark this answers is not in the document]
Comment (jwbos) [reply iacrcc/iacrcc.cls:577]: The point is not to check / validate the email address. We want to make sure this looks somewhat like a valid string used for an email address and not that a comma has been forgotten in the \addauthor macro. 
IMO authors should not simply provide firstname dot lastname, if they provide an e-mail address it should resemble a real one.

File: iacrcc/iacrcc.cls [hunk 2]
```diff
@@ -552,6 +552,44 @@
...
+  \regex_match:nnTF {^\d{4}-\d{4}-\d{4}-\d{3}[Xx\d]$} { #1 }
+  { \prg_return_true:  }
+  { \prg_return_false: }
+}
+\cs_new_eq:NN \IfValidOrcid \iacr_check_orcid:nTF
+
+\prg_new_conditional:Npnn \iacr_check_inst:n #1 { T, F, TF }
+{
+  \cs_generate_variant:Nn \regex_match:nnTF {nV}
+  % An institute index should be a number, multiple institutes are separated by a comma:
```
[the remark this answers is not in the document]
Comment (jwbos) [reply iacrcc/iacrcc.cls:570]: Exactly, same point as above. This is just a basic sanity check not a comprehensive validation.

**Pull request merged**